class/practice5.1/MegaData.h

```diff
@@ -0,0 +1,30 @@
+#ifndef MEGADATA_H
+#define MEGADATA_H
```

Comment:
Some minor comments:
1. We can use pragma once - it works fine in most of the modern compilers so we can assume that we are using such a compiler as we are writing code on the latest C++23. ifndef idiom is still used in case of old or custom compilers.
2. We can move MegaDataPool to a separate file - in general we can reuse and extend it later to be more generic and support other types.
3. Another small note on your implementation of the pool (just to think about general problematics of the topic) - we are returning  pointer to data therefore technically someone can delete the underlying memory. As an alternative we can return some reference or std::ref type to avoid this issue. Or we can rely on the user to be responsible enough which in some cases may be a solution too.